docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,270 @@
+
+Performing and managing these operations individually for a long list of software and configuration items would be cumbersome,
+especially when performed on a large fleet of devices as each operation will have to be monitored and managed separately.
+Grouping them together into a single operation reduces that overhead considerably as you just have one operation to monitor
+instead of multiple.
+
+**Q2: Why not model each device profile update as a firmware update that includes the desired software and configurations as well**
+
+Although this is a more robust approach, it is not feasible on all kinds of devices,
+especially the ones that does not support delta firmware updates.
+On such devices, pushing the entire firmware binary for each iterative change would considerably increase the binary size overhead.
+
+# Requirements
+
+* Ability to control the order of execution of the operations defined in the command input
+* Ability to apply 
+
+# Device profile capability
+
+A device that supports device profile operation must declare that capability by publishing the following message:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile' '{
+  "name": "prod-profile",
+  "version: "v1"
+}'
+```
+
+If the current profile information is not known upfront, publish an empty JSON (`{}`) instead.
+The firmware information, supported software and configuration types must be declared separately with their respective capability messages.
+
+# Device profile command
+
+Once the `device_profile` capability is declared, the device can receive `device_profile` commands
+by subscribing to `<root>/<device-topic-id>/cmd/device_profile/+` MQTT topics.
+For example, subscribe to the following topic for the `main` device:
+
+```sh te2mqtt formats=v1
+tedge mqtt sub 'te/device/main///cmd/device_profile/+'
+```
+
+A `device_profile` command with `id` "1234" is triggered as follows:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile/1234' '{
+  "status": "init",
+  "name": "prod-profile",
+  "version": "v2",
+  "profile": [
+    {
+      "operation": "firmware_update",
+      "skip": false,
+      "payload": {
+        "name": "core-image-tedge-rauc",
+        "remoteUrl": "https://abc.com/some/firmware/url",
+        "version": "20240430.1139"
+      }
+    },
+    {
+      "operation": "software_update",
+      "skip": false,
+      "payload": {
+        "updateList": [
+          {
+            "type": "apt",
+            "modules": [
+              {
+                "name": "c8y-command-plugin",
+                "version": "latest",
+                "action": "install"
+              },
+              {
+                "name": "collectd",
+                "version": "latest",
+                "type": "apt",
+                "action": "install"
+              }
+            ]
+          }
+        ]
+      }
+    },
```

Comment:
Based on the feedback from last round, I've updated the input format moving away from the previous flat structure and kept it in-line with what @reubenmiller had originally proposed. The definitely helps with payload consistency although I'm not a bit unhappy with the level of nesting that's there.

Reply:
>  I'm not a bit unhappy with the level of nesting that's there.

You mean *"I'm a bit unhappy "*? Don't you?

Level of nesting doesn't matter as this is read by a machine.